syntax = "proto3";
package sym;

service Sym {
    // Our SayHello rpc accepts HelloRequests and returns HelloReplies
    rpc SayHello (HelloRequest) returns (HelloReply);
    // Convert md formula returning sympy function
    rpc ConvertMdFormula (ConvertMdRequest) returns (ConvertMdReply);
}

message HelloRequest {
    // Request message contains the name to be greeted
    string name = 1;
}

message HelloReply {
    // Reply contains the greeting message
    string message = 1;
}

enum ValueType {
    NUMBER = 0;
    EXPR = 1;
}

message ConvertMdRequest {
    // Request contains the md fomula
    string md = 1;
}

message ConvertMdReply {
    // Reply contains the converted and substituded formula
    string sym = 1;
}
